pkg/datapath/linux/node.go

```diff
@@ -127,6 +129,59 @@ func updateTunnelMapping(oldCIDR, newCIDR *cidr.CIDR, oldIP, newIP net.IP, first
 	}
 }
 
+// updateIPCacheExternalVxlanMapping is called
+// with external VXLAN device encapsulation mode enabled.
+// the CIDR, Endpoint, MAC are of external VXLAN device
+func updateIPCacheExternalVxlanMapping() {
+	var (
+		worldID              uint32 = 2 //identity
+		encryptKey           uint8  = 0 //no encrypt
+		externalVxlanCIDR    *cidr.CIDR
+		exernalVxlanEndpoint net.IP
+		externalVtepMAC      mac.MAC
+		err                  error
+	)
+
+	for i := 0; i < len(option.Config.ExternalVxlanEndpoint); i++ {
+		externalVxlanCIDR, err = cidr.ParseCIDR(option.Config.ExternalVxlanCIDR[i])
+		if err != nil {
+			log.WithError(err).WithFields(logrus.Fields{
+				"allocCIDR": option.Config.ExternalVxlanCIDR[i],
+			}).Error("bpf: Unable to parse external VXLAN CIDR")
+			continue
```

Comment:
Typically we will handle this kind of input validation super early on as part of `pkg/option/config.go:(DaemonConfig).Validate()`. That way, this code deep in the datapath sections can trust that the input was already parsed & structured correctly and doesn't need to worry about this kind of error condition.

Reply:
good idea, I will check it out